### ISSUE TITLE
bug: ESLint VSCode not working out of the box

### ISSUE THREAD

**Issue by smblee** [t3-oss/create-t3-turbo#1128]
### Provide environment information

  System:
    OS: macOS 14.4
    CPU: (10) arm64 Apple M1 Pro
    Memory: 55.73 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.15.1 - ~/.local/share/mise/installs/node/20.15/bin/node
    npm: 10.7.0 - ~/.local/share/mise/installs/node/20.15/bin/npm
    pnpm: 9.6.0 - ~/.local/share/mise/installs/node/20.15/bin/pnpm

### Describe the bug

try adding something that would trigger ESlint error like

`const EXPO_COOKIE_NAME = "__acme-expo-redirect-state" as any;`

This isn't getting caught by VSCode.

![CleanShot 2024-07-28 at 22 34 25@2x](https://github.com/user-attachments/assets/fae8085c-df80-42c9-ae6e-ab9a7b395d19)

But it does get caught in `pnpm lint`
![CleanShot 2024-07-28 at 22 26 49@2x](https://github.com/user-attachments/assets/86107b6c-09c3-49c8-ae7c-8e3868c073d9)

Getting the following error in ESLint output.
![CleanShot 2024-07-28 at 22 36 19@2x](https://github.com/user-attachments/assets/8c8b2d4f-eed7-4409-89d2-eb1863c8df78)

### Link to reproduction

.

### To reproduce

try adding something that would trigger ESlint error like

`const EXPO_COOKIE_NAME = "__acme-expo-redirect-state" as any;`

**Comment by w4jo** [comment]
![Screenshot 2024-07-29 at 3 46 46 PM](https://github.com/user-attachments/assets/6576d5e2-0443-4337-acd0-21e6f1f09f95)

In this case, everything works fin for me.
I think `import.meta.dirname` return `undefined`.

**Comment by dBianchii** [comment]
> ![Screenshot 2024-07-29 at 3 46 46 PM](https://private-user-images.githubusercontent.com/44121169/353097786-6576d5e2-0443-4337-acd0-21e6f1f09f95.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIzODc4MTIsIm5iZiI6MTcyMjM4NzUxMiwicGF0aCI6Ii80NDEyMTE2OS8zNTMwOTc3ODYtNjU3NmQ1ZTItMDQ0My00MzM3LWFjZDAtMjFlNmYxZjA5Zjk1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzMxVDAwNTgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4OWFmNGUxYWZiY2E2MDlhMDU1NmU4NGFmYWFiYTg3Zjk2NWFmZTg5NjY1MmRmYmQ4MWIzODJmMWMwMDMzNjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0KEBP0xbv4g96DD0TiZ43EaZr97sdnbjVUwol3uI3lk)
> 
> In this case, everything works fin for me. I think `import.meta.dirname` return `undefined`.

Oh wow... Thank you. Can confirm this fixes it.
Just comment this line and its imports, run pnpm clean:workspaces, restart eslint and typescript server, run pnpm i and then restart the servers again.

**Comment by w4jo** [comment]
Well, yes, there is a bit of work there.
You can close the issue now ☺️

**Comment by w4jo** [comment]
Sure, I had to do something else to get around this issue. I’ll try to figure out a solution and create a PR for it, if I have some time.

**Comment by smblee** [comment]
```
import { fileURLToPath } from "node:url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

export default tseslint.config(
  // Ignore files not tracked by VCS and any config files
  includeIgnoreFile(path.join(__dirname, "../../.gitignore")),
  ```
  
  This is what i did to resolve

**Comment by smblee** [comment]
Ok according to @juliusmarminge 

> What Node version are you on? import.meta.dirname is available on 20.11 and later: [nodejs.org/api/esm.html#importmetadirname](https://nodejs.org/api/esm.html#importmetadirname)

And that was indeed the issue.

Apparently ESLint plugin defaults to using whatever Node that was shipped with. 

I set this option in vscode settings and now it works.

```
 "eslint.runtime": "node"
 ```

**Comment by juliusmarminge** [comment]
> I set this option in vscode settings and now it works.
>  "eslint.runtime": "node"

This seems like what we want, from the extension:

`eslint.runtime - use this setting to set the path of the node runtime to run ESLint under. [Use "node"](https://github.com/microsoft/vscode-eslint/issues/1233#issuecomment-815521280) if you want to use your default system version of node.`
